vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -513,6 +378,41 @@ float ConnectionBoxMapLookahead::get_expected_cost(
     }
 }
 
+static int manhattan_distance(const t_rr_node &node, int x, int y) {
+  int node_center_x = (node.xhigh() + node.xlow()) / 2;
+  int node_center_y = (node.yhigh() + node.ylow()) / 2;
+  return abs(node_center_x - x) + abs(node_center_y - y);
+}
+
+void ConnectionBoxMapLookahead::find_start_inodes_for_segment_types(int num_segments) {
+    auto &device_ctx = g_vpr_ctx.device();
+    auto &rr_nodes = device_ctx.rr_nodes;
+    const int halo = std::min(device_ctx.grid.width(), device_ctx.grid.height()) / 4;
+    const int center_x = device_ctx.grid.width() / 2;
```

Comment:
We already talked about why the center of the grid is actually not any "interesting" location, so why are you sampling near the center of the grid, rather than the center of the segment type?

Reply:
Just to see how well it works. I'll try calculating a bounding box per type.